pandas/_libs/groupby.pyx

```diff
@@ -253,18 +253,16 @@ def group_cumsum(numeric[:, ::1] out,
                         t = accum[lab, j] + y
                         compensation[lab, j] = t - accum[lab, j] - y
                         accum[lab, j] = t
-                        out[i, j] = accum[lab, j]
+                        out[i, j] = t
```

Comment:
Doubt this affects compiled result, but may as well to not depend on a smart compiler